pkg/project/registry/projectrequest/delegated/delegated.go

```diff
@@ -42,7 +42,6 @@ func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 
 	project := &projectapi.Project{}
 	project.ObjectMeta = projectRequest.ObjectMeta
-	project.DisplayName = projectRequest.DisplayName
```

Comment:
This line needs to be the following:

```
project.Annotations["displayName"]=projectRequest.DisplayName
```

This fixes the issue I reported with $ osc new-project foo --display-name="Something" not setting the display name.